### ISSUE TITLE
`cucumber-chef setup` should terminate EC2 instance on failure

### ISSUE THREAD

**Issue by zts** [Atalanta/cucumber-chef#9]
I'm currently having a problem (unrelated to cucumber-chef) where my new EC2 instance is unreachable.  This causes `cucumber-chef setup` to fail, but the instance is left in the "running" state.

```$ cucumber-chef setup
Provisioning cucumber-chef test lab platform.
Waiting for server...................
Instance ID: i-009be476 ; IP Address 46.137.60.137
Platform provisioned.  Run cucumber-chef project to get started.
WARNING: Failed to connect to ec2-46-137-60-137.eu-west-1.compute.amazonaws.com -- Errno::ETIMEDOUT: Operation timed out - connect(2)
/Users/zts/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/net/http.rb:2303:in`error!': 404 "Not Found" (Net::HTTPServerException)
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/chef-0.10.0/lib/chef/rest.rb:237:in `block in api_request'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/chef-0.10.0/lib/chef/rest.rb:288:in`retriable_rest_request'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/chef-0.10.0/lib/chef/rest.rb:218:in `api_request'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/chef-0.10.0/lib/chef/rest.rb:114:in`get_rest'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/chef-0.10.0/lib/chef/node.rb:603:in `load'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/cucumber-chef-1.0.0/lib/cucumber/chef/provisioner.rb:74:in`tag_node'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/cucumber-chef-1.0.0/lib/cucumber/chef/provisioner.rb:31:in `bootstrap_node'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/cucumber-chef-1.0.0/bin/cucumber-chef:78:in`setup'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/thor-0.14.6/lib/thor/invocation.rb:118:in`invoke_task'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/thor-0.14.6/lib/thor.rb:263:in `dispatch'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/thor-0.14.6/lib/thor/base.rb:389:in`start'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/gems/cucumber-chef-1.0.0/bin/cucumber-chef:139:in `<top (required)>'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/bin/cucumber-chef:19:in`load'
    from /Users/zts/.rvm/gems/ruby-1.9.2-p180@chef/bin/cucumber-chef:19:in `<main>'

```

Ideally, `setup` would attempt to terminate the running instance before returning control to the shell.
Minimally, a warning should be printed, advising the user to manually terminate the instance in the AWS management console.
```

**Comment by danicuki** [comment]
I have the same problem. You said the the problem is unrelated to cucumber-chef. What it is related to? Do you know how can I solve it?

**Comment by danicuki** [comment]
Maybe it is something related to the chosen linux image. What image are you using?

**Comment by zts** [comment]
Sorry, I certainly could have put more explanation into this report.
For me, at least, the problem was that I had never used Amazon EC2 before.  When you first start using EC2, the default "security group" does not allow you to ssh (tcp port 22) to the machines you create.  This obviously causes a problem for cucumber-chef.

I've started a fix for this in Issue #19, but until that's available you can fix this yourself through the AWS EC2 management console.
- On the EC2 tab, click "Security Groups" on the left hand side.
- Click the "default" group in the list.
- Click the "Inbound" tab, in the pane below the list of groups.
- Select "SSH" from the drop-down list, and click "Add rule" (restricting to your netblocks, if you like)
- Click "apply rule changes"

After that, things worked correctly.

**Comment by danicuki** [comment]
After changing the SSH configuration in Amazon, I get a different error:

WARNING: Failed to connect to ec2-50-17-146-127.compute-1.amazonaws.com -- Errno::ECONNREFUSED: Connection refused - connect(2)
/Users/danicuki/.rvm/rubies/ruby-1.9.2-p0/lib/ruby/1.9.1/net/http.rb:2295:in `error!': 404 "Not Found" (Net::HTTPServerException)
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/rest.rb:237:in`block in api_request'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/rest.rb:288:in `retriable_rest_request'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/rest.rb:218:in`api_request'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/rest.rb:114:in `get_rest'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/node.rb:603:in`load'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/lib/cucumber/chef/provisioner.rb:74:in `tag_node'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/lib/cucumber/chef/provisioner.rb:30:in`bootstrap_node'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/bin/cucumber-chef:63:in `setup'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor/task.rb:22:in`run'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor/invocation.rb:118:in `invoke_task'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor.rb:263:in`dispatch'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor/base.rb:389:in `start'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/bin/cucumber-chef:123:in`<top (required)>'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/bin/cucumber-chef:19:in `load'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/bin/cucumber-chef:19:in`<main>'

SSH works if I do it by hand with:
ssh -i my_key.pem ec2-user@ec2-50-17-146-127.compute-1.amazonaws.com

Do you think there's something to do with the ec2-user configuration?

**Comment by danicuki** [comment]
I've just realized that if you choose an image other than ubuntu, it won't work for cucumber-chef, once it has hard coded ssh username to ubuntu. Now I change for a ubuntu image, but it still get stucked after this:

Waiting for server............
Instance ID: i-fff4459e ; IP Address 50.19.3.254
Platform provisioned.  Run cucumber-chef project to get started.

Then I hit CTRL+C and get:

/Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/net-ssh-multi-1.0.1/lib/net/ssh/multi/session_actions.rb:37:in `join': Interrupt
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/net-ssh-multi-1.0.1/lib/net/ssh/multi/session_actions.rb:37:in`block in sessions'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/net-ssh-multi-1.0.1/lib/net/ssh/multi/session_actions.rb:37:in `each'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/net-ssh-multi-1.0.1/lib/net/ssh/multi/session_actions.rb:37:in`sessions'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/net-ssh-multi-1.0.1/lib/net/ssh/multi/session_actions.rb:81:in `open_channel'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/knife/ssh.rb:161:in`ssh_command'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/knife/ssh.rb:320:in `run'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/chef-0.10.2/lib/chef/knife/bootstrap.rb:147:in`run'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/lib/cucumber/chef/provisioner.rb:29:in `bootstrap_node'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/bin/cucumber-chef:63:in`setup'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor/invocation.rb:118:in`invoke_task'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor.rb:263:in `dispatch'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/thor-0.14.6/lib/thor/base.rb:389:in`start'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/gems/cucumber-chef-1.0.3/bin/cucumber-chef:123:in `<top (required)>'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/bin/cucumber-chef:19:in`load'
    from /Users/danicuki/.rvm/gems/ruby-1.9.2-p0/bin/cucumber-chef:19:in `<main>'

**Comment by leinster** [comment]
Fixed with #19.
